lib/util.js

```diff
@@ -425,6 +436,28 @@ function formatValue(ctx, value, recurseTimes) {
     return ctx.stylize('null', 'null');
   }
 
+  if (ctx.complexity > 0) {
+    if (ctx.stop === true) {
+      const name = getConstructorName(value) || value[Symbol.toStringTag];
+      return ctx.stylize(`[${name || 'Object'}]`, 'special');
+    }
+    if (ctx.time === undefined) {
+      ctx.time = process.hrtime();
+    } else if (getClockTime(ctx.time) > 1e3) {
```

Comment:
Is this necessary?  Leaving out wall time tracking results in a simpler algorithm.

Reply:
The point is: without this it is hard to tell how much time this will actually need on different CPUs. Using the wall time is future proof by always delivering the best result with a best effort mechanism.

Reply:
What I mean is, is tracking any kind of time necessary?

What you call complexity, I would call budget. You start with a fixed budget and spend it until it's gone.

Since you have that kind of logic in place already, and since it's deterministic and clock agnostic, why also track time?

Reply:
Budget is indeed a good description. I am just not sure if this would be understood by looking at a option named `budget`.

Relying on the budget completely without checking the wall time as well means I can measure the time it takes on my machine and estimate a arbitrary time limit that should not be exceeded in the default case (e.g. 1000-1250 ms as currently done). However, we also have Raspberry PIs and similar and their CPUs are much much slower and the same budget would get spend slower and therefore block the event loop longer. Our CPUs in general get faster over time with new inventions though, so the current default case for a fast machine would soon drop and the budget is spend faster. Meaning the maximum time the event loop would be allowed to be blocked would  drop to e.g., 250ms instead of the current limit.

Using both together makes sure that it will always be around in the 1000-1250ms range. That is the reason why I would like to have the combination of both, even though it is not fully deterministic after spending the complete budget.

Reply:
That's a fair point.  Have you considered splitting it out into two separate options, one for budget/complexity and one for wall time?  Or are you saying that you feel a single option is best?

Reply:
I feel a single option is best for two reasons: 

- We already have to many options in util inspect and the more we add, the more complicated it gets for the user. I believe we should just have very solid defaults while allowing the user to also change some things that are definitely necessary in some cases.
- I do not see a necessity for having a fine grained control over this. The option is meant to prevent the event loop from being blocked to long while being able to also opt out of that and to stricten the rule further. The question is: is it necessary for the user to allow util.inspect to bail out before one second passed? I personally doubt that.